### PR TITLE
BluetoothThread refactor

### PR DESCRIPTION
**Pull request by dati91**
Minor things like rename or for replacement.

### CODE REVIEW

File: components/net/bluetooth_thread.rs [hunk 1]
```diff
@@ -23,13 +23,28 @@ const CHARACTERISTIC_ERROR: &'static str = "No characteristic found";
...
+        const EMPTY                       = 0b000000000,
+        const BROADCAST                   = 0b000000001,
+        const READ                        = 0b000000010,
+        const WRITE_WITHOUT_RESPONSE      = 0b000000100,
+        const WRITE                       = 0b000001000,
+        const NOTIFY                      = 0b000010000,
+        const INDICATE                    = 0b000100000,
+        const AUTHENTICATED_SIGNED_WRITES = 0b001000000,
+        const RELIABLE_WRITE              = 0b010000000,
+        const WRITABLE_AUXILIARIES        = 0b100000000,
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:37]: binary nums :heart:

File: components/net/bluetooth_thread.rs [hunk 2]
```diff
@@ -23,13 +23,28 @@ const CHARACTERISTIC_ERROR: &'static str = "No characteristic found";
...
+bitflags! {
+    flags Flags: u32 {
+        const EMPTY                       = 0b000000000,
+        const BROADCAST                   = 0b000000001,
+        const READ                        = 0b000000010,
+        const WRITE_WITHOUT_RESPONSE      = 0b000000100,
+        const WRITE                       = 0b000001000,
+        const NOTIFY                      = 0b000010000,
+        const INDICATE                    = 0b000100000,
+        const AUTHENTICATED_SIGNED_WRITES = 0b001000000,
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:35]: maybe `..._WRITE`

File: components/net/bluetooth_thread.rs [hunk 3]
```diff
@@ -44,10 +59,7 @@ pub trait BluetoothThreadFactory {
 impl BluetoothThreadFactory for IpcSender<BluetoothMethodMsg> {
     fn new() -> IpcSender<BluetoothMethodMsg> {
         let (sender, receiver) = ipc::channel().unwrap();
-        let adapter = match BluetoothAdapter::init() {
-            Ok(a) => Some(a),
-            Err(_) => None,
-        };
+        let adapter = BluetoothAdapter::init().ok();
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:62]: I hope there will be some error handling

File: components/net/bluetooth_thread.rs [hunk 4]
```diff
@@ -170,14 +185,13 @@ impl BluetoothManager {
...
         let device_id = match self.service_to_device.get_mut(service_id) {
             Some(d) => d.clone(),
             None => return None,
         };
-        // Update cache
-        self.get_gatt_services(adapter, &device_id);
-        check_cache!(self.cached_services, service_id);
+        self.get_and_cache_gatt_services(adapter, &device_id);
+        return_if_cached!(self.cached_services, service_id);
         None
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:235]: Is it even possible to reach this line? Can `get_and_cache_gatt_services` fail?

**Comment by mmatyas** [comment]
Nice patch!

File: components/net/bluetooth_thread.rs (hunk 4 above)
Comment (dati91) [inline components/net/bluetooth_thread.rs:235]: yes, if there no such service

File: components/net/bluetooth_thread.rs (hunk 3 above)
Comment (dati91) [inline components/net/bluetooth_thread.rs:62]: there are, this is why we use Option<>

File: components/net/bluetooth_thread.rs (hunk 2 above)
Comment (dati91) [inline components/net/bluetooth_thread.rs:35]: http://git.kernel.org/cgit/bluetooth/bluez.git/tree/doc/gatt-api.txt#n140

**Comment by zakorgy** [comment]
lgtm

**Comment by dati91** [comment]
@mmatyas r?

File: components/net/bluetooth_thread.rs (hunk 2 above)
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:35]: Ok :+1:

**Comment by mmatyas** [comment]
r+

**Pull request merged**